### PR TITLE
Update links about referencing fields in processors

### PR DESCRIPTION
**Pull request by hariso**
### Description

Part of #1422.

Related to https://github.com/ConduitIO/conduit-site/pull/50 (it creates the page to which we link).

### Quick checks:

- [ ] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [ ] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [ ] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/plugin/processor/builtin/impl/filter.go [hunk 1]
```diff
@@ -35,10 +35,12 @@ func (p *filterProcessor) Specification() (sdk.Specification, error) {
...
 		Name:    "filter",
 		Summary: "Acknowledges all records that get passed to the filter.",
-		Description: `Acknowledges all records that get passed to the filter, so the records will be filtered out if 
-the condition provided to the processor is evaluated to "true".
+		Description: `Acknowledges all records that get passed to the filter, so
+the records will be filtered out if the condition provided to the processor is
+evaluated to ` + "`true`" + `.
 
-**Important:** Make sure to add a condition to this processor, otherwise all records will be filtered out.`,
+**Important:** Make sure to add a [condition](https://conduit.io/docs/processors/conditions)
```
Comment (raulb) [inline pkg/plugin/processor/builtin/impl/filter.go:42]: I see this page is still not done, so I left a comment in https://github.com/ConduitIO/conduit/issues/1422 to not forget url is this one.

File: pkg/plugin/processor/builtin/internal/exampleutil/specs/avro.decode.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
       },
       "field": {
         "default": ".Payload.After",
-        "description": "The field that will be encoded.",
+        "description": "The field that will be decoded.\n\nFor more information about the format, see [Referencing fields](https://conduit.io/docs/processors/referencing-fields).",
```
Comment (raulb) [inline pkg/plugin/processor/builtin/internal/exampleutil/specs/avro.decode.json:23]: great gatch.

File: pkg/plugin/processor/builtin/internal/exampleutil/specs/base64.decode.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "parameters": {
       "field": {
         "default": "",
-        "description": "Field is the target field, as it would be addressed in a Go template (e.g. `.Payload.After.foo`).\nNote that it is not allowed to base64 decode the `.Position` field.",
+        "description": "Field is the reference to target field. Note that it is not allowed to\nbase64 decode the `.Position` field.\n\nFor more information about the format, see [Referencing fields](https://conduit.io/docs/processors/referencing-fields).",
```
Comment (raulb) [inline pkg/plugin/processor/builtin/internal/exampleutil/specs/base64.decode.json:11]:
```suggestion
        "description": "Field is the reference to the target field. Note that it is not allowed to\nbase64 decode the `.Position` field.\n\nFor more information about the format, see [Referencing fields](https://conduit.io/docs/processors/referencing-fields).",
```

**Review by raulb — APPROVED** [review]
Approving though I left a suggestion of something missing. Once that's done, this is ready to ship 👍

**Comment by lovromazgon** [comment]
I sneaked in another change - I renamed all `examples_test.go` and `examples_exporter_test.go` files (used to generate processor specs), they now have the prefix `z_` which makes them alphabetically the last file in the folder, meaning that `go:generate` will process them last. Why is this good? They depend on the code that gets produced by paramgen, so this way we make sure that paramgen runs first.

File: pkg/plugin/processor/builtin/impl/filter.go (hunk 1 above)
Comment (lovromazgon) [reply pkg/plugin/processor/builtin/impl/filter.go:42]: Good catch, thanks!

**Pull request merged**